subworkflows/local/prepare_genome.nf

```diff
@@ -0,0 +1,72 @@
+//
+// Prepare reference genome files
+//
+
+include { BISMARK_GENOMEPREPARATION   } from '../../modules/nf-core/bismark/genomepreparation/main'
+include { BWAMETH_INDEX               } from '../../modules/nf-core/bwameth/index/main'
+include { SAMTOOLS_FAIDX              } from '../../modules/nf-core/samtools/faidx/main'
+
+workflow PREPARE_GENOME {
+
+    main:
+    versions = Channel.empty()
+
+    fasta = Channel.empty()
+    bismark_index = Channel.empty()
+    bwameth_index = Channel.empty()
+    fasta_index = Channel.empty()
+
+    // FASTA, if supplied
+    if (params.fasta) {
```

Comment:
I like the idea of avoiding referencing params in subworkflows, should this be factored into a channel as well?